### DISCUSSION TITLE
adding-authentication-to-a-blazor-server-app-using-auth0/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#360] [Announcements]
# adding-authentication-to-a-blazor-server-app-using-auth0/

In this post I show how to add authentication to a Blazor Server application using Auth0.

https://andrewlock.net/adding-authentication-to-a-blazor-server-app-using-auth0/

**Comment by dracan** [comment]
Amazing! Thank you for spending the time to write this - I've just had this requirement, and following your blog post worked perfectly! 🙏 And I didn't hit the NameClaimType issue either - when I got to that section, it was correctly showing my name.

**Reply by andrewlock** [reply]
Awesome, always great to hear! 😀😎

**Reply by dracan** [reply]
What should call the `/account/accessdenied` page? I had to add an additional `OpenIdConnectEvents` event registration...

```csharp
            OnAccessDenied = context =>
            {
                context.Response.Redirect("/account/accessdenied");
                context.HandleResponse();
                return Task.CompletedTask;
            }
```

[a comment by an automated account is not shown]

**Reply by dracan** [reply]
I wondered that too. I had to add this...

https://github.com/andrewlock/blog-comments/discussions/360#discussioncomment-6017487

[a comment by an automated account is not shown]

**Comment by ThinhNgVhust** [comment]
Hi author, I am Thinh from VietNam and i am trying to implement Blazor(Server app) Single Sign On using SAML2. 
Could you write a post about this approach?
